File: Semester 3/MAP/Seminar10/Seminar10/factory/TaskContainerFactory.cs
using Seminar10.container;
using Seminar10.factory;
using Seminar10.strategy;

namespace Seminar10.factory;

public class TaskContainerFactory : Factory
{
    private TaskContainerFactory()
    {
        
    }

    private static TaskContainerFactory taskContainerFactory = new TaskContainerFactory();
    
    public static TaskContainerFactory getInstance()
    {
        return taskContainerFactory;
    }
    
    public Container createContainer(ContainerStrategy strategy)
    {
        switch (strategy)
        {
            case ContainerStrategy.LIFO:
                return new StackContainer();
            case ContainerStrategy.FIFO:
                return new QueueContainer();
            default:
                return null;
        }
    }
}

File: Semester 3/MAP/Sem1_pb1_MAP/src/factory/TaskContainerFactory.java
package factory;

public class TaskContainerFactory implements Factory{
    private static TaskContainerFactory instance ;
    private TaskContainerFactory(){}

    public static TaskContainerFactory getInstance(){
        if(instance == null){
            instance = new TaskContainerFactory();
        }
        return instance;
    }

    @Override
    public Container createContainer(Strategy strategy) {
        if(strategy.equals(Strategy.LIFO)){
            return new StackContainer();
        }
        else {
            return new QueueContainer(); //return new Q
        }
    }
}
